LocaleEmulator/Hooks/Gdi32Hook.cpp

```diff
@@ -160,17 +223,27 @@ HGDIOBJ NTAPI LeGetStockObject(LONG Object)
 
             GlobalData->HookRoutineData.Gdi32.StockObjectInitialized = TRUE;
         }
-    }
+    }*/
 
     LOOP_ONCE
     {
-        if (Object > countof(GlobalData->HookRoutineData.Gdi32.StockObject))
+        if (Object >= countof(GlobalData->HookRoutineData.Gdi32.StockObject))
             break;
 
-        StockObject = GlobalData->HookRoutineData.Gdi32.StockObject[Object];
+        auto& StockObject = GlobalData->HookRoutineData.Gdi32.StockObject[Object];
```

Comment:
`GlobalData->HookRoutineData.Gdi32.StockObject` is never filled. Lines 219-222 should be run when this method is first called with a `FONT_*` argument, just to fill the cache.

Reply:
Or store StockObject into the cache after running Line 243.

Reply:
> Or store StockObject into the cache after running Line 243.

Yeah, this is what I did.  I modified line 233 to have `StockObject` become a reference of an element in `GlobalData->HookRoutineData.Gdi32.StockObject[]`.

Reply:
Understood, thanks!